tools/building.py

```diff
@@ -863,7 +863,19 @@ def closure_compiler(filename, pretty, advanced=True, extra_closure_args=None):
   # Specify input file relative to the temp directory to avoid specifying non-7-bit-ASCII path names.
   for e in CLOSURE_EXTERNS:
     args += ['--externs', e]
-  args += user_args
+  # Clients may specify extern files or js files like:
+  # --closure-args=--externs=user_externs.js
+  # We need to split this argument so the file stands alone as its own argument.
+  # This allows us to convert any relative path to an absolute path in run_closure_cmd.
+  for c in user_args:
+    if c.startswith('--externs='):
+      args.append('--externs')
+      args.append(c[len('--externs='):])
+    elif c.startswith('--js='):
+      args.append('--js')
+      args.append(c[len('--js='):])
```

Comment:
How about:

```
# Ensure filenames are specified as standalone arguments so that the
# code below can take care of converting any relative paths.
If any(c.startswith(p) for p in ('--externs=', '--js=')):
   args.extend(c.lsplit('=', 1))
else:
   args.append(c)
```

Reply:
Actually I wonder if its better to improve the code below to handle the single argument form.. to limit this hack to just one place in the codebase?

Reply:
Something like:

```
for i in range(len(cmd)):
  for prefix in ('--externs', '--js'):	
    if cmd[i] == prefix	
      cmd[i + 1] = move_to_safe_7bit_ascii_filename(cmd[i + 1])	
      break
   elif cmd[i].startswith(prefix + '='):
      filename = cmd[i].lsplit('=', 1)[1]
      cmd[i] = '='.join(prefix, move_to_safe_7bit_ascii_filename(filename)	
 ```
```

Reply:
Done.